jina/orchestrate/deployments/config/k8slib/kubernetes_deployment.py

```diff
@@ -89,16 +93,29 @@ def get_deployment_yamls(
     elif image_name_uses_after:
         template_name = 'deployment-uses-after'
 
-    yamls = [
-        kubernetes_tools.get_yaml(
-            'configmap',
+    if monitoring:
+        service_yaml = kubernetes_tools.get_yaml(
+            'service_monitoring',
             {
                 'name': name,
+                'target': name,
                 'namespace': namespace,
-                'data': env,
+                'port': port,
+                'type': 'ClusterIP',
+                'port_monitoring': port_monitoring,
             },
-        ),
-        kubernetes_tools.get_yaml(
+        )
+
+        service_monitor_yaml = kubernetes_tools.get_yaml(
+            'service_monitor',
```

Comment:
what is this service?

Reply:
The ServiceMonitor is not a kubernetes service but  a prometheus custom object. It allows prometheus to automatically discover new service to monitor

Reply:
but then the user must have some Operator or CRD installed?